providers/implementations/keymgmt/ecx_kmgmt.c

```diff
@@ -706,15 +707,17 @@ static int ecx_key_pairwise_check(const ECX_KEY *ecx, int type)
 static int ecx_validate(const void *keydata, int selection, int type, size_t keylen)
 {
     const ECX_KEY *ecx = keydata;
-    int ok = 0;
+    int ok = keylen == ecx->keylen;
 
     if (!ossl_prov_is_running())
         return 0;
 
-    assert(keylen == ecx->keylen);
+    if ((selection & ECX_POSSIBLE_SELECTIONS) == 0)
+        return 1; /* nothing to validate */
 
-    if ((selection & ECX_POSSIBLE_SELECTIONS) != 0)
-        ok = 1;
+    if (!ok) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_ALGORITHM_MISMATCH);
```

Comment:
return 0?

Reply:
I can add that although it does not change much.

Reply:
I actually had it there once - that explains the curly brackets that would not be needed otherwise.